### PR TITLE
QuantityLog 100%

### PR DESCRIPTION
**Pull request by MartyEternal**
QuantityLog stuff is all done for now.

- Fixed the issue where the item.quantity_current wasn't matching up with the quantity_log.change number(s)
- Updated the details that the user can view regarding these changes
- Added the QuantityLog details as a table in the History Log page. Intended to have them shown side-by-side.

### CODE REVIEW

File: main_app/forms.py [hunk 1]
```diff
@@ -32,5 +32,5 @@ def __init__(self, *args, **kwargs):
 class QuantityChangeForm(forms.ModelForm):
 	class Meta:
 		model = QuantityLog
-		fields = ['item', 'change']
+		fields = ['change']
```
Comment (MartyEternal) [inline main_app/forms.py:35]: Removing 'item' means users don't have to choose the item from the list

File: main_app/migrations/0009_quantitylog_previous_count_alter_item_description_and_more.py
Comment (MartyEternal) [inline main_app/migrations/0009_quantitylog_previous_count_alter_item_description_and_more.py:1]: Don't forget to migrate!

File: main_app/models.py [hunk 2]
```diff
@@ -7,6 +7,7 @@
 class Category(models.Model):
     name = models.CharField(max_length=100)
     description = models.TextField(max_length=250)
+    is_deleted = models.BooleanField(default=False)
```
Comment (MartyEternal) [inline main_app/models.py:10]: This is_deleted boolean field is in preparation for the reworking of the delete function

File: main_app/models.py [hunk 3]
```diff
@@ -17,11 +18,12 @@ def get_absolute_url(self):
     # Create your models here.
 class Item(models.Model):
     name = models.CharField(max_length=100)
-    quantity_current = models.IntegerField()
+    quantity_current = models.IntegerField(default=0)
     quantity_min = models.IntegerField(default = 1)
     quantity_max = models.IntegerField()
-    description = models.TextField(max_length=250)
+    description = models.TextField(max_length=1000)
```
Comment (MartyEternal) [inline main_app/models.py:24]: Was copy/pasting a description from Amazon. Realised that 250 characters isn't really enough for a decent description of the item.

File: main_app/models.py [hunk 4]
```diff
@@ -32,6 +34,7 @@ def get_absolute_url(self):
 class QuantityLog(models.Model):
     item = models.ForeignKey(Item, on_delete=models.CASCADE)
     date_log = models.DateTimeField(auto_now_add=True)
+    previous_count = models.IntegerField()
```
Comment (MartyEternal) [inline main_app/models.py:37]: Had to add another field to the QuantityLog model to make it all work properly

File: main_app/templates/main_app/history_log.html
Comment (MartyEternal) [inline main_app/templates/main_app/history_log.html:1]: history_log.html now includes both the HistoryLog and QuantityLog

File: main_app/views.py [hunk 5]
```diff
@@ -146,9 +146,9 @@ def items_details(request, item_id):
     categories_item_doesnt_have = Category.objects.exclude(id__in=id_list)
     last_quantity_changes = QuantityLog.objects.filter(item=item).order_by('-date_log')[:5]
 
-    for log in last_quantity_changes:
-        previous_logs = QuantityLog.objects.filter(item=item, date_log__lt=log.date_log).order_by('-date_log')
-        log.previous_count = previous_logs[0].change if previous_logs.exists() else item.quantity_current - log.change
+    # for log in last_quantity_changes:
```
Comment (MartyEternal) [inline main_app/views.py:149]: Forgot to delete this part of the code. Had it commented out.

File: main_app/views.py [hunk 6]
```diff
@@ -157,14 +157,18 @@ def items_details(request, item_id):
             quantity_log.item = item
             quantity_log.user = request.user
             quantity_log.previous_count = item.quantity_current
-            item.quantity_current += quantity_log.change
+            item.quantity_current = quantity_log.change
```
Comment (MartyEternal) [inline main_app/views.py:160]: Had to change this line here as it was the main reason for the item.quantity_current to not show the same number that was input into the change form.

File: main_app/views.py [hunk 7]
```diff
@@ -203,8 +207,9 @@ def unassoc_category(request, item_id, category_id):
 
 # history log stuff
 def history_log(request):
-    history = HistoryLog.objects.all()
-    return render(request, 'main_app/history_log.html', {'history': history})
+    history = HistoryLog.objects.all().order_by('-date_log')
```
Comment (MartyEternal) [inline main_app/views.py:210]: Just added QuantityLog to the History Log page.

**Review by MartyEternal — COMMENTED** [review]
All checked. Need second pair of eyes please.

**Review by Exiled184 — APPROVED** [review]
code looks good

**Pull request merged**